pystac_client/stac_api_io.py

```diff
@@ -0,0 +1,225 @@
+                                                    preserve_dict=preserve_dict)
+            result._stac_io = self
+            return result
+
+        if info.object_type == pystac.STACObjectType.COLLECTION:
+            return pystac_client.CollectionClient.from_dict(d,
+                                                            href=href,
+                                                            root=root,
+                                                            migrate=False,
+                                                            preserve_dict=preserve_dict)
+
+        if info.object_type == pystac.STACObjectType.ITEM:
+            return pystac.Item.from_dict(d,
+                                         href=href,
+                                         root=root,
+                                         migrate=False,
+                                         preserve_dict=preserve_dict)
+
+        raise ValueError(f"Unknown STAC object type {info.object_type}")
+
+    def get_pages(self, url, method='GET', parameters={}) -> Iterator[Dict]:
+        """Iterator that yields dictionaries for each page at a STAC paging endpoint, e.g., /collections, /search
+
+        Yields
+        -------
+        Dict : JSON content from a single page
+        """
+        page = self.read_json(url, method=method, parameters=parameters)
+        yield page
+
+        next_link = next((link for link in page.get('links', []) if link['rel'] == 'next'), None)
+        while next_link:
+            link = Link.from_dict(next_link)
+            page = self.read_json(link, parameters=parameters)
+            yield page
+
+            # get the next link and make the next request
+            next_link = next((link for link in page.get('links', []) if link['rel'] == 'next'),
+                             None)
+
+    def conforms_to(self, conformance_class: ConformanceClasses) -> bool:
+        """Whether the API conforms to the given standard. This method only checks against the ``"conformsTo"``
+        property from the API landing page and does not make any additional calls to a ``/conformance`` endpoint
+        even if the API provides such an endpoint.
+
+        Parameters
+        ----------
+        key : str
+            The ``ConformanceClasses`` key to check conformance against.
+
+        Returns
+        -------
+        bool
+            Indicates if the API conforms to the given spec or URI.
+        """
+
+        # Conformance of None means ignore all conformance as opposed to an
+        #  empty array which would indicate the API conforms to nothing
+        if self._conformance is None:
+            return True
+
+        uris = CONFORMANCE_URIS.get(conformance_class.name, None)
+
+        if uris is None:
+            raise Exception(f"Invalid conformance class {conformance_class}")
+
+        if not any(uri in uris for uri in self._conformance):
+            raise NotImplementedError(f"{conformance_class} not supported")
```

Comment:
Is there a reason to raise an error instead of just returning `False`?

Reply:
I implemented it as a bool, then added back the exception raising.